### PR TITLE
Inline links styling

### LINKED ISSUE #424: Inline links - styling
**Is your feature request related to a problem? Please describe.**
Styles of inline links do not meet UX requirements.

**Describe the solution you'd like**
Inline links should have the following styles:
- color:  `#0F556C` bluejay
- border-bottom: 1px solid `#0F556C` bluejay
- font-family: same as text

background-color: not applicable

**Additional Context**
  - All links open in current tab
  - No bold or italic (i.e. the formatting is removed) - does not block this ticket, done in #481
  - This issue may be worked on at the same time as any PR in the following group: #424, #425, #426, #427

**Testing**
- Cypress test
- Tab through to link
- Open via keyboard should open in current tab
- Check superscript/subscript characters in links (dependent on #454)

**Screenshot of designs**
inline link

### LINKED ISSUE #425: Inline links - visited styles
**Is your feature request related to a problem? Please describe.**
Visited styles of inline links do not meet UX requirements.

**Describe the solution you'd like**
Inline links should have the following styles in a visited state:
- border-bottom: 1px solid `#5C5752` pebble
- color: `#5C5752` pebble

background color: not applicable

**Additional context**
- This issue may be worked on at the same time as any PR in the following group: #424, #425, #426, #427

**Testing**
- Force visited state
- Cypress test
- Check superscript/subscript characters in links (dependent on #454)

**Screenshot of designs**
image

### LINKED ISSUE #426: Inline links - hover styles
**Is your feature request related to a problem? Please describe.**
Hover styles of inline links do not meet UX requirements.

**Describe the solution you'd like**
Inline links should have the following styles in a hover state:
- background-color: `#C3DEE7` bluejay light
- border: 2px solid `#C3DEE7` bluejay light
- border-bottom-color: 1px solid `#0F556C` bluejay
- color: `#0F556C` bluejay

**Additional context**
- This issue may be worked on at the same time as any PR in the following group: #424, #425, #426, #427

**Testing**
- Force hover state
- Cypress test

**Screenshot of designs**
image

### LINKED ISSUE #427: Inline links - focus styles
**Is your feature request related to a problem? Please describe.**
Focus styles of inline links do not meet UX requirements.

**Describe the solution you'd like**
Inline links should have the following styles in a focussed state:
- background-color: `#0F556C` bluejay
- border: 2px solid `#0F556C` bluejay
- border-bottom: 1px solid `#F5F3F1` oat-light
- color: `#F5F3F1` oat-light

**Additional context**
- This issue may be worked on at the same time as any PR in the following group: #424, #425, #426, #427

**Testing**
- Force focused state
- Tab through links and ensure can tab through to link
- Cypress tests

**Screenshot of designs**
image

### PR DESCRIPTION
**Pull request by jtart**
Resolves #424, Resolves #425, Resolves #426, Resolves #427

_Adds inlink styles that conform to UX requirements._

- [x] Tests added for new features
- [x] Test engineer approval

### CODE REVIEW

File: src/app/lib/constants/styles.js [hunk 1]
```diff
@@ -6,7 +6,10 @@ export const C_EBON = '#222222';
 export const C_POSTBOX = '#B80000';
 export const C_STORM = '#404040';
 export const C_WHITE = '#FFFFFF';
-
+export const C_BLUEJAY = '#0f556c';
```
Comment (sareh) [inline src/app/lib/constants/styles.js]: Could you please make this all caps, for consistency with other constants?

File: src/app/components/Text/index.stories.jsx [hunk 2]
```diff
@@ -15,4 +15,7 @@ storiesOf('Text', module)
   ))
   .add('inline code', () => (
     <Text text="This is text with `some inline code`." />
+  ))
+  .add('inline linke', () => (
```
Comment (pjlee11) [inline src/app/components/Text/index.stories.jsx]: `linke` 🤔

File: src/app/components/Text/index.test.jsx [hunk 3]
```diff
@@ -26,4 +26,9 @@ describe('Text', () => {
     'should render inline-code',
     <Text text="This is text with `some inline code`." />,
   );
+
+  shouldMatchSnapshot(
+    'should render an inline link',
+    <Text text="This is text that contains an [inline link](https://www.bbc.com/news) inside it.." />,
```
Comment (pjlee11) [inline src/app/components/Text/index.test.jsx]: `it..`

**Comment by jtart** [comment]
image
default link styles
image
hover styles
image
focus styles
image
visited styles

**Comment by pjlee11** [comment]
Suggestion: if UX would like the border bottom to not be slanted on the edges we could try using padding and a border bottom instead of a transparent border all round.

I quickly POC'd this using `padding: 2px 2px 0;` in place of the 2px border and the outcome was as expected:

image

**Review by sareh — CHANGES_REQUESTED** [review]
It would be better to have `InlineLink` as its own component and imported in within the Text component. This is to give visibility to those using the component library that the InlineLink is self-contained and reusable. 
The bottom border has angled points, which do not match the design and this will likely be picked up by UX in their review.

**Review by pjlee11 — APPROVED** [review]
Nice work 👌

**Review by sareh — APPROVED** [review]
👍 Looks great! Thanks for pulling out `InlineLink` to be a separate component.

**Comment by jamesbrumpton** [comment]
LGTM

**Pull request merged**